src/codeu/chat/server/Model.java

```diff
@@ -135,15 +146,133 @@ public void add(Message message) {
     return messageByText;
   }
   
+  @Override
+  public void changeAuthority(ConversationUuid conversation, Uuid targetUser, byte authorityByte){
+	StoreAccessor<Uuid, ConversationHeader> convos = this.conversationById();
+    ConversationHeader chat = convos.first(conversation.root());
+    HashMap<Uuid, Byte> accessMap = this.getPermissionMap(chat);
+    accessMap.put(targetUser, authorityByte);
+    authority.put(chat.id, accessMap);
+  }
+
+  public void initializeAuthority(Uuid conversation, Uuid targetUser, byte authorityByte)
+  {
+    StoreAccessor<Uuid, ConversationHeader> convos = this.conversationById();
+    ConversationHeader chat = convos.first(conversation);
+    HashMap<Uuid, Byte> accessMap = this.getPermissionMap(chat);
+    accessMap.put(targetUser, authorityByte);
+    authority.put(chat.id, accessMap);
+  }
+  
+  @Override
+  public boolean isMember(ConversationHeader conversation,Uuid targetUser){
+    HashMap<Uuid, Byte> accessMap = this.getPermissionMap(conversation);
+    if(!accessMap.containsKey(targetUser)){
+  	  accessMap.put(targetUser, USER_TYPE_MEMBER);
+  	  authority.put(conversation.id, accessMap);
+  	  return true;
+  	} else {
+  	  byte user = accessMap.get(targetUser);
+  	  if ((user | USER_TYPE_BANNED) == USER_TYPE_BANNED)
+  	  	return false;
+  	  return true;
+    }
+  }
+   
+  @Override
+  public boolean isMember(ConversationUuid conversation,Uuid targetUser){
+	StoreAccessor<Uuid, ConversationHeader> convos = this.conversationById();
+	ConversationHeader chat = convos.first(conversation.root());
+	HashMap<Uuid, Byte> accessMap = this.getPermissionMap(chat);
+	if(!accessMap.containsKey(targetUser)){
+	  accessMap.put(targetUser, USER_TYPE_MEMBER);
```

Comment:
Same issue here. I don't think you should be adding membership status when checking if a user is a member.